torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1356,6 +1359,35 @@ def _free_full_params_and_use_local_shard(params_to_free):
                         _free_full_params_and_use_local_shard(currently_local_params)
                         self.training_state = TrainingState_.IDLE
 
+    def named_parameters(
+        self,
+        prefix: str = "",
+        recurse: bool = True,
+    ) -> Iterator[Tuple[str, torch.nn.Parameter]]:
+        """
+        Overrides :meth:`torch.nn.Module.named_parameters()` to intercept
+        parameter names and remove all occurrences of the FSDP-specific
+        flattened parameter prefix when inside the :meth:`summon_full_params`
+        context manager.
+        """
+        # Determine which logic to use based on the context at call time
+        if not hasattr(self, "training_state") or \
+                self.training_state != TrainingState_.SUMMON_FULL_PARAMS:
+            for param_name, param in torch.nn.Module.named_parameters(
+                self, prefix=prefix, recurse=recurse
+            ):
+                # Do not modify the behavior if not in `summon_full_params()`
+                yield (param_name, param)
+        else:
+            for param_name, param in torch.nn.Module.named_parameters(
+                self, prefix=prefix, recurse=recurse,
+            ):
+                # Remove any instances of "_fsdp_wrapped_module._fpw_module.";
+                # there can be multiple in the case of nested FSDP modules
+                fsdp_prefix = FSDP_WRAPPED_MODULE + "." + FPW_MODULE + "."
```

Comment:
Please create a FSDP_PREFIX global variable just like FSDP_WRAPPED_MODULE. This is actually a constant.